File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/model/WeekId.java
package com.salesianos.triana.BeElite.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Service;

import java.io.Serializable;
@AllArgsConstructor
@NoArgsConstructor
@Embeddable
@Getter
@Setter
public class WeekId implements Serializable {

    @Column(name = "week_number")
    private Long week_number;

    @Column(name = "week_name")
    private String week_name;

    @ManyToOne
    @JoinColumn(name = "program_id")
    private Program program;

    public static WeekId of(Long id, String week_name, Program program){
        return new WeekId(id,week_name,program);
    }
}

File: frontend/be_elite/lib/models/Session/Athlete Sessions/week_id.dart
class WeekId {
	int? weekNumber;
	String? weekName;
	String? programId;

	WeekId({this.weekNumber, this.weekName, this.programId});

	factory WeekId.fromJson(Map<String, dynamic> json) => WeekId(
				weekNumber: json['week_number'] as int?,
				weekName: json['week_name'] as String?,
				programId: json['program_id'] as String?,
			);

	Map<String, dynamic> toJson() => {
				'week_number': weekNumber,
				'week_name': weekName,
				'program_id': programId,
			};
}
